calico-enterprise/networking/gateway-api.mdx

```diff
@@ -0,0 +1,325 @@
+---
+description: Enable support for the Kubernetes Gateway API.
+---
+
+# Gateway API
+
+## Big picture
+
+Enable support for the Kubernetes Gateway API.
+
+## Value
+
+{{prodname}} includes support for the Kubernetes Gateway API, which allows advanced routing to services in a cluster, including weighted or blue-green load balancing.
+
+## Concepts
+
+### Gateway API
+
+The Gateway API is an official Kubernetes API for advanced routing to services in a cluster.  To read about its use cases, structure and design, please see [the official docs](https://gateway-api.sigs.k8s.io/).
+
+### Envoy Gateway
+
+Several implementations of the Gateway API are available, one of which is the [Envoy Gateway](https://gateway.envoyproxy.io/).  {{prodname}} integrates the Envoy Gateway implementation in order to provide support for the Gateway API.
+
+### Access into a cluster from outside
+
+The Gateway API only provides access into a cluster from outside when the cluster is _also_ provisioned to support Kubernetes Services with `type: LoadBalancer`.  When a Gateway is configured, {{prodname}} creates a Deployment that does the actual work of routing and load balancing, etc., and a Service with `type: LoadBalancer` that fronts that Deployment.  If the cluster has a `type: LoadBalancer` provider, it will then allocate an IP outside the cluster and arrange for requests to that IP to be forwarded to the Gateway Service.
+
+Managed Kubernetes services like AKS, EKS and GKE include a `type: LoadBalancer` provider that automatically integrates with Azure, AWS and GCP respectively.  On-prem clusters and non-managed clusters in the cloud need to set up their own `type: LoadBalancer` support.
+
+## Before you begin
+
+**Unsupported**
+
+- Windows
+
+## How to
+
+- [Enable Gateway API support](#enable-gateway-api-support)
+- [Use the Gateway API](#use-the-gateway-api)
+- [Disable Gateway API support](#disable-gateway-api-support)
+
+### Enable Gateway API support
+
+To enable Gateway API support, create a `GatewayAPI` resource with name `tigera-secure`:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: operator.tigera.io/v1
+kind: GatewayAPI
+metadata:
+  name: tigera-secure
+EOF
+```
+
+Shortly after this you will see that Gateway API resources are now available:
+
+```bash
+kubectl api-resources | grep gateway.networking.k8s.io
+```
+
+And also that there is a GatewayClass resource corresponding to the Envoy Gateway implementation included in {{prodname}}:
+
+```bash
+kubectl get gatewayclass -o yaml
+```
+
+### Use the Gateway API
+
+Now you can use the Gateway API in any way that the API supports.  When configuring a Gateway resource, be sure to specify `gatewayClassName: tigera-gateway-class`.
```

Comment:
Should we be calling it `tigera-gateway`? `-class` is redundant.

Reply:
We could do.  But in this case I think the `-class` is helpful, on balance.  There are so many things that are called "gateway": the API, the controller, the Gateway objects, and the GatewayClass.  So here I think it's helpful to be more explicit.

Reply:
We don't add suffixes to any resource we create in the operator, like storage classes, deployments, services, ... I think we should be consistent in naming things.